qiskit/synthesis/linear_phase/cx_cz_depth_lnn.py

```diff
@@ -185,37 +194,63 @@ def _apply_phase_to_nw_circuit(n, phase_schedule, seq, swap_plus):
 
         p = phase_schedule[j, k]
 
-        if (j, k) not in swap_plus:
-            cir.cx(w1, w2)
-
-        cir.cx(w2, w1)
+        if use_dag:
+            if (j, k) not in swap_plus:
+                cir.apply_operation_back(CXGate(), (qreg[w1], qreg[w2]), check=False)
+            cir.apply_operation_back(CXGate(), (qreg[w2], qreg[w1]), check=False)
+        else:
+            if (j, k) not in swap_plus:
+                cir.cx(w1, w2)
+            cir.cx(w2, w1)
 
         if p % 4 == 0:
             pass
         elif p % 4 == 1:
-            cir.sdg(w2)
+            if use_dag:
+                cir.apply_operation_back(SdgGate(), (qreg[w2],), check=False)
+            else:
+                cir.sdg(w2)
         elif p % 4 == 2:
-            cir.z(w2)
+            if use_dag:
+                cir.apply_operation_back(ZGate(), (qreg[w2],), check=False)
+            else:
+                cir.z(w2)
         else:
-            cir.s(w2)
-
-        cir.cx(w1, w2)
+            if use_dag:
+                cir.apply_operation_back(SGate(), (qreg[w2],), check=False)
+            else:
+                cir.s(w2)
+        if use_dag:
+            cir.apply_operation_back(CXGate(), (qreg[w1], qreg[w2]), check=False)
+        else:
+            cir.cx(w1, w2)
 
     for i in range(n):
         p = phase_schedule[n - 1 - i, n - 1 - i]
         if p % 4 == 0:
             continue
         if p % 4 == 1:
-            cir.sdg(i)
+            if use_dag:
+                cir.apply_operation_back(SdgGate(), (qreg[i],), check=False)
+            else:
+                cir.sdg(i)
         elif p % 4 == 2:
-            cir.z(i)
+            if use_dag:
+                cir.apply_operation_back(ZGate(), (qreg[i],), check=False)
+            else:
+                cir.z(i)
         else:
-            cir.s(i)
+            if use_dag:
+                cir.apply_operation_back(SGate(), (qreg[i],), check=False)
+            else:
+                cir.s(i)
 
     return cir
 
 
-def synth_cx_cz_depth_line_my(mat_x: np.ndarray, mat_z: np.ndarray) -> QuantumCircuit:
+def synth_cx_cz_depth_line_my(
```

Comment:
Added `use_dag` to the function because it's used in  `synth_clifford_depth_lnn`.